nms/app/packages/magmalte/app/state/feg/EquipmentState.js

```diff
@@ -136,3 +233,50 @@ async function getActiveFegGatewayId(
     return '';
   }
 }
+
+type FetchProps = {
+  networkId: network_id,
+  id?: gateway_id,
+  enqueueSnackbar?: (
+    msg: string,
+    cfg: EnqueueSnackbarOptions,
+  ) => ?(string | number),
+};
+
+/**
+ * Fetches and returns the list of gateways under the federation network or
+ * the specific gateway if the id is provided.
+ *
+ * @param {network_id} networkId: Id of the federation network.
+ * @param {gateway_id} id id of the federation gateway.
+ * @param {(msg, cfg,) => ?(string | number),} enqueueSnackbar Snackbar to display error
+ * @returns returns a list of the federation gateways or a federation gateway with the given id.
+ */
+export async function FetchFegGateways(props: FetchProps) {
+  const {networkId, id, enqueueSnackbar} = props;
+  if (id !== undefined && id !== null) {
```

Comment:
You should be able to do a direct truthiness check on `id` here. A zero-length `gateway_id` is nonsense anyways.

https://www.sitepoint.com/javascript-truthy-falsy/